geode-core/src/main/java/org/apache/geode/internal/cache/control/HeapMemoryMonitor.java

```diff
@@ -554,12 +579,12 @@ private String createCriticalThresholdLogMessage(MemoryEvent event, String event
   }
 
   public long getMaxMemory() {
-    return heapUsageMonitor.getMaxMemory();
+    return heapUsageProvider.getMaxMemory();
   }
 
   @TestOnly
-  public NotificationListener getHeapUsageMonitor() {
-    return (MemoryPoolMXBeanHeapUsageProvider) heapUsageMonitor;
+  public NotificationListener getHeapUsageProvider() {
+    return (MemoryPoolMXBeanHeapUsageProvider) heapUsageProvider;
```

Comment:
Should this not be cast to a NotificationListener?

Reply:
This method only exists to get an existing test: MemoryMonitorIntegrationTest to work.
But I think getHeapUsageProvider will go away in the near future. This level of detail should be tested by a unit test on MemoryPoolMXBeanHeapUsageProvider not this integration test.